samplekotlin/src/main/java/com/example/chattutorial/ChannelActivity3.kt

```diff
@@ -95,7 +96,10 @@ class ChannelActivity3 : AppCompatActivity() {
 
         // Observe typing events and update typing header depending on its state.
         lifecycleScope.launchWhenStarted {
-            ChatClient.instance().globalState.typingUpdates.collect {
```

Comment:
`GlobalState` should not be used here. It would provide typing updates not for a particular channel but for all active channels.

In order to get updates for a particular active channel we have to watch it first and collect the resulting `ChannelState`.